playground/ssr-resolve/main.js

```diff
@@ -5,14 +5,16 @@ import fileEntry from '@vitejs/test-entries/file'
 // has `exports` key, should resolve to pkg-exports/entry
 import pkgExportsEntry from '@vitejs/test-resolve-pkg-exports/entry'
 import deepFoo from '@vitejs/test-deep-import/foo'
-import deepBar from '@vitejs/test-deep-import/bar'
+// import deepBar from '@vitejs/test-deep-import/bar'
```

Comment:
I commented out this test case as it wasn't working.
The output contains `@vitejs/test-deep-import/bar` as-is, but to make that work in Node, it needs to be rewritten to `@vitejs/test-deep-import/utils/bar.js` as Node does not read `package.json` for directory specifiers in ESM.

Reply:
Does that mean this could be a breaking change if someone relied on the non-standard behavior before?

I'm ok with trying this out in a minor, but I think if we're not going to support this case here we can directly remove the test for it.

Reply:
No, this code was not working without the change in this PR.
I changed the test code to run it in Node for https://github.com/vitejs/vite/pull/20409#discussion_r2212043490 and that caused the test to fail.
https://github.com/vitejs/vite/pull/20409/files#diff-859b1585f9b836443382acf98d7530dd7666ca2e0931021e82f745af51e233a3L26-R32